### PR TITLE
Added benf\neo\models\BlockType::icon

### PR DESCRIPTION
**Pull request by leevigraham**
Initial PR to add an icon property to blocks.

I'm still trying to figure out how to implement `autosuggest` aliases when entering the url.

Screen Shot 2022-05-22 at 12 57 56 pm

Icon UI needs some work… this just shows they're rendering:

Screen Shot 2022-05-22 at 1 37 04 pm

### CODE REVIEW

File: client/src/configurator/BlockTypeSettings.js [hunk 1]
```diff
@@ -165,13 +171,24 @@ export default Settings.extend({
...
       name: descriptionInputName,
       label: Craft.t('neo', 'Description'),
-      // instructions: Craft.t('neo', 'How you’ll refer to this block type in the templates.'),
       required: false,
       value: this.getDescription(),
       errors: errors.handle
     })
     $descriptionInput.find('input').attr('data-neo-bts', 'input.description')
 
+    const $iconInput = Craft.ui.createTextField({
```
Comment (leevigraham) [inline client/src/configurator/BlockTypeSettings.js:180]: @brandonkelly is there a way to enable alias auto suggest functionality when creating text fields in javascript?

**Comment by ttempleton** [comment]
I don't mind having the option for icons on the existing block type buttons, but if this is intended as another step towards #569, another option is to limit icons to the block type picker, by having a field setting that controls whether the existing block type buttons or the block type picker appears also controlling whether the block type icon fields are displayed on the block type settings.

**Comment by leevigraham** [comment]
Yep it's intended as a step towards #569. I just threw them in the code for now to test the alias parsing and to show everything was hooked up properly.

File: client/src/configurator/BlockTypeSettings.js (hunk 1 above)
Comment (ttempleton) [reply client/src/configurator/BlockTypeSettings.js:180]: If it isn't possible in JS, I can look at moving the block type settings rendering to PHP.

Comment (leevigraham) [reply client/src/configurator/BlockTypeSettings.js:180]: Moving to PHP twig would be interesting… probably easier to understand.

Comment (ttempleton) [reply client/src/configurator/BlockTypeSettings.js:180]: Fair enough 😅 I'll have a look at that next week.

Comment (ttempleton) [reply client/src/configurator/BlockTypeSettings.js:180]: I've moved it to PHP/Twig in the [enhancement/block-type-settings-rendering](https://github.com/spicywebau/craft-neo/tree/enhancement/block-type-settings-rendering) branch. Testing with an autosuggest field seems to be working.

**Pull request closed without merging**